texttospeech/synthesize_file.rb

```diff
@@ -0,0 +1,92 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def synthesize_text_file text_file:
+  # [START tts_synthesize_text_file]
+  require "google/cloud/text_to_speech"
+
+  client = Google::Cloud::TextToSpeech.new
+
+  text       = File.open(text_file, "r") { |file| file.read }
+  input_text = { text: text }
+
+  # Note: the voice can also be specified by name.
+  # Names of voices can be retrieved with client.list_voices
+  voice = {
+    language_code: "en-US",
+    ssml_gender:   "FEMALE"
+  }
+
+  audio_config = { audio_encoding: "MP3" }
+
+  response = client.synthesize_speech input_text, voice, audio_config
+
+  # The response's audio_content is binary.
+  File.open("output.mp3", "wb") do |file|
+    # Write the response to the output file.
+    file.write(response.audio_content)
+  end
+
+  puts "Audio content written to file 'output.mp3'"
+  # [END tts_synthesize_text_file]
+end
+
+def synthesize_ssml_file ssml_file:
+  # [START tts_synthesize_ssml_file]
+  require "google/cloud/text_to_speech"
+
+  client = Google::Cloud::TextToSpeech.new
+
+  ssml       = File.open(ssml_file, "r") { |file| file.read }
+  input_text = { ssml: ssml }
+
+  # Note: the voice can also be specified by name.
+  # Names of voices can be retrieved with client.list_voices
+  voice = {
+    language_code: "en-US",
+    ssml_gender:   "FEMALE"
+  }
+
+  audio_config = { audio_encoding: "MP3" }
+
+  response = client.synthesize_speech input_text, voice, audio_config
+
+  # The response's audio_content is binary.
+  File.open("output.mp3", "wb") do |file|
+    # Write the response to the output file.
+    file.write(response.audio_content)
+  end
+
+  puts "Audio content written to file 'output.mp3'"
+  # [END tts_synthesize_ssml_file]
+end
+
+if __FILE__ == $PROGRAM_NAME
+  command   = ARGV.shift
+  file_path = ARGV.shift
+
+  if command == "--text"
```

Comment:
Update command to follow other samples:

For example, instead of `--text` use `text`.